fix: add net.transport to http semcov (#1824)
open-telemetry/opentelemetry-specification: specification/trace/semantic_conventions/http.md

used to look up the `net.peer.ip` that is actually connected to.
In that case it is strongly recommended to set the `net.peer.name` attribute in addition to `http.host`.

> Note that `net.peer.port` is only applicable for ip transports and therefore only required if IP transport is used.

Reviewer: ```suggestion
If a transport not based on TCP is used, and no http.host is available, then (in absence of `http.host` and `http.url`) leaving out `net.peer.port` is allowed, but `net.transport` must be set instead.
```

Author: I used a different wording as I don't want to make transport describing entries depend on http ones.

Reviewer: But without that, it reads like transport is always required if a non-TCP transport is used. Which would be OK for me, but then we'd need to add a it as a `ref:` attribute to the common HTTP attributes with `required: conditional: "If different from ip_tcp"`.

Author: Sorry I don't understand what you mean. `net.transport` is as requires as `net.peer.name` and it is mentioned in the yaml in the same way. So why is there a `ref:` needed for the one but not for the other?

Reviewer: The `ref` would be needed if we want to add the `required: conditional:` I suggested.

used to look up the `net.peer.ip` that is actually connected to.
In that case it is strongly recommended to set the `net.peer.name` attribute in addition to `http.host`.

> If transport is not based on IP `net.peer.port` is not applicable. Instead `net.transport` must be set.

Reviewer: ```suggestion
If transport is not based on IP `net.peer.port` is not applicable. Instead `net.transport` must be set (if neither `http.url` nor `http.host` are set).
```

Author: As said I would avoid to mix http concerns with transport concerns. Please note that you need `net.transport` to correctly interpret `net.peer.name` if it is not guranteed to be IP.

Reviewer: This document is about the HTTP semantic conventions here though. The general net semantic conventions are already described to that effect in <https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#general-network-connection-attributes>

In any case, it needs to be clear if, for HTTP spans, net.transport is required only if http.host/http.url is not set or for any non-TCP transport. Currently this is unclear to me.

Author: But this is unrelated to this PR. Even before this PR it was not mentioned anywhere that e.g. `net.peer.name` and `net.peer.port` is not required if http.host/http.url is present.

Except if listing them as a separate line is enough then this is also done now.

It is strongly recommended to set `http.server_name` to allow associating requests with some logical server entity.

> Note that `net.host.port` is only applicable for ip transports and therefore only required if IP transport is used.

Reviewer 2: ```suggestion
Note that `net.host.port` is only applicable for IP transports and therefore only required if IP transport is used.
```

It is strongly recommended to set `http.server_name` to allow associating requests with some logical server entity.

> Note that `net.host.port` is only applicable for IP transports and therefore only required if IP transport is used.

Reviewer: I would use the same wording as above
